### PR TITLE
Keep Samba binaries for dependency nmblookup

### PR DESCRIPTION
**Pull request by razzeee**
Cherry-pick from @14mRh4X0r targeting the beta branch https://github.com/flathub/tv.kodi.Kodi/pull/174

### CODE REVIEW

**Comment by wsnipex** [comment]
this isn't needed for Nexus, see https://github.com/xbmc/xbmc/pull/22068

**Comment by razzeee** [comment]
The update might still be worth it?

**Comment by wsnipex** [comment]
yes, but not as is. Nmblookup is not needed.
Why is this switching back to autotools? 
Why is json added, when we use --without-json?

**Comment by razzeee** [comment]
fair, will look into that

File: modules/samba/samba.json [hunk 1]
```diff
@@ -7,17 +7,15 @@
         "./buildtools/bin/waf install"
     ],
     "cleanup": [
-        "/bin",
```
Comment (Erick555) [inline modules/samba/samba.json:10]: If `nmblookup` isn't needed then `cleanup`  should stay untouched since there are tins of unnecessary stuff there.

Comment (razzeee) [reply modules/samba/samba.json:10]: Whoops, forgot, that these were multiple commits

**Comment by 14mRh4X0r** [comment]
There's a reason I split my PR up in two commits, 4bea5715 should apply cleanly, and should be all and everything that's needed.

Now for the questions raised:

> Why is this switching back to autotools?

This is not switching to autotools, but rather Samba's own configure script and makefile [as per instructions](https://wiki.samba.org/index.php/Build_Samba_from_Source#configure): calling `waf` explicitly doesn't work anymore.

> Why is json added, when we use --without-json?

It seems to be a build-time dependency, as can be seen in the logs for [build 5667](https://buildbot.flathub.org/#/builders/6/builds/5667)

**Pull request closed without merging**